builtin/mainmenu/tab_settings.lua

```diff
@@ -15,497 +15,504 @@
+				local current_value = get_current_value(setting)
+				core.setting_setbool(setting.name, not core.is_yes(current_value))
+				core.setting_save()
+				return true
+			else
+				list_enter = true
+			end
 		else
-			core.setting_set("enable_shaders", fields["cb_shaders"])
+			return true
 		end
-		return true
 	end
-	if fields["cb_bumpmapping"] then
-		core.setting_set("enable_bumpmapping", fields["cb_bumpmapping"])
-	end
-	if fields["cb_generate_normalmaps"] then
-		core.setting_set("generate_normalmaps", fields["cb_generate_normalmaps"])
-	end
-	if fields["cb_parallax"] then
-		core.setting_set("enable_parallax_occlusion", fields["cb_parallax"])
+	
+	if fields["btn_edit"] or list_enter then
+		local setting = settings[selected_setting]
+		if setting.type ~= "category" then
+			local edit_dialog = dialog_create("change_setting", create_change_setting_formspec,
+					handle_change_setting_buttons)
+			edit_dialog:set_parent(this)
+			this:hide()
+			edit_dialog:show()
+		end
 		return true
 	end
-	if fields["cb_waving_water"] then
-		core.setting_set("enable_waving_water", fields["cb_waving_water"])
+
+	if fields["btn_restore"] then
+		local setting = settings[selected_setting]
+		if setting.type ~= "category" then
+			core.setting_set(setting.name, setting.default)
+			core.setting_save()
+			core.update_formspec(this:get_formspec())
+		end
 		return true
 	end
-	if fields["cb_waving_leaves"] then
-		core.setting_set("enable_waving_leaves", fields["cb_waving_leaves"])
+
+	if fields["btn_change_keys"] then
+		core.show_keys_menu()
+		return true
 	end
-	if fields["cb_waving_plants"] then
-		core.setting_set("enable_waving_plants", fields["cb_waving_plants"])
+
+	if fields["cb_tech_settings"] then
+		core.setting_set("main_menu_technical_settings", fields["cb_tech_settings"])
+		core.setting_save()
+		core.update_formspec(this:get_formspec())
 		return true
 	end
 
-	if fields["sb_gui_scaling"] then
-		local event = core.explode_scrollbar_event(fields["sb_gui_scaling"])
+	return false
+end
 
-		if event.type == "CHG" then
-			local tosave = string.format("%.2f",scrollbar_to_gui_scale(event.value))
-			core.setting_set("gui_scaling",tosave)
-			return true
+local function create_minetest_conf_example()
+	local result = "#    This file contains a list of all available settings and their default value for minetest.conf\n" ..
+			"\n" ..
+			"#    By default, all the settings are commented and not functional.\n" ..
+			"#    Uncomment settings by removing the preceding #.\n" ..
+			"\n" ..
+			"#    minetest.conf is read by default from:\n" ..
+			"#    ../minetest.conf\n" ..
+			"#    ../../minetest.conf\n" ..
+			"#    Any other path can be chosen by passing the path as a parameter\n" ..
+			"#    to the program, eg. \"minetest.exe --config ../minetest.conf.example\".\n" ..
+			"\n" ..
+			"#    Further documentation:\n" ..
+			"#    http://wiki.minetest.net/\n" ..
+			"\n"
+
+	for _, entry in pairs(settings) do
```

Comment:
ipairs is bad as it doesn't preserve order, and if you automatically create the .conf.example it will bloat up and obfuscate the diff.